### PR TITLE
Add 1.30 Release Team Permissions

### PR DESCRIPTION
**Pull request by katcosgrove**
Initial PR to add RT Lead and Subteam Leads for v1.30.

### CODE REVIEW

File: config/kubernetes/sig-release/teams.yaml [hunk 1]
```diff
@@ -79,7 +80,6 @@ teams:
     - justaugustus # Release
     - justinsb # AWS / Cluster Lifecycle
     - k8s-release-robot # Release
```
Comment (Priyankasaggu11929) [inline config/kubernetes/sig-release/teams.yaml:82]: failing test - https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/org/4643/pull-org-test-all/1742181606761697280
> config_test.go:268: The team milestone-maintainers in org Kubernetes has non-admins listed as maintainers; these users should be in the members list instead: katcosgrove

```suggestion
    - k8s-release-robot # Release
    - katcosgrove # v1.30 RT Lead
```

File: config/kubernetes/sig-release/teams.yaml [hunk 2]
```diff
@@ -17,9 +17,10 @@ teams:
     description: Contributors who can use `/milestone` or `/status` commands on
       issues/PRs and have triage access to the kubernetes/enhancements repo
     maintainers:
+    - katcosgrove # v1.30 RT Lead
```
Comment (Priyankasaggu11929) [inline config/kubernetes/sig-release/teams.yaml:20]:
```suggestion
```

File: config/kubernetes/sig-release/teams.yaml [hunk 3]
```diff
@@ -305,8 +306,8 @@ teams:
       release-team:
         description: Members of the current Release Team and subproject owners.
         maintainers:
+        - katcosgrove # 1.30 RT Lead
         - palnabarun # subproject owner (1.21 RT Lead)
```
Comment (Priyankasaggu11929) [inline config/kubernetes/sig-release/teams.yaml:310]: Previous Release team leads (for currently maintained releases) stays in the list

```suggestion
        - palnabarun # subproject owner (1.21 RT Lead)
        - Priyankasaggu11929 # subproject owner (1.29 RT Lead)
```

File: config/kubernetes/sig-release/teams.yaml [hunk 4]
```diff
@@ -305,8 +306,8 @@ teams:
       release-team:
         description: Members of the current Release Team and subproject owners.
         maintainers:
+        - katcosgrove # 1.30 RT Lead
```
Comment (Priyankasaggu11929) [inline config/kubernetes/sig-release/teams.yaml:309]:
```suggestion
```

File: config/kubernetes/sig-release/teams.yaml [hunk 5]
```diff
@@ -305,8 +306,8 @@ teams:
...
         - palnabarun # subproject owner (1.21 RT Lead)
-        - Priyankasaggu11929 # 1.29 RT Lead
         members:
         - cpanato # subproject owner / Technical Lead
         - gracenng # subproject owner (1.28 RT Lead)
```
Comment (Priyankasaggu11929) [inline config/kubernetes/sig-release/teams.yaml:313]: >  config_test.go:268: The team release-team in org Kubernetes has non-admins listed as maintainers; these users should be in the members list instead: katcosgrove


```suggestion
        - gracenng # subproject owner (1.28 RT Lead)
        - katcosgrove # 1.30 RT Lead 
```

**Review by Priyankasaggu11929 — CHANGES_REQUESTED** [review]

**Comment by Priyankasaggu11929** [comment]
@katcosgrove @gracenng, how about we merge the existing `ci-signal` and `release-team-bug-triage` GH teams, to a new `release-team-release-signal` team (to align with our new release team structure change, and avoid duplicating same team members across 2 teams)

```suggestion
          release-team-release-signal:
            description: Members of the Release Signal team for the current release cycle.
            members:
            - pacoxu # 1.30 Release Signal Lead
            # - X # 1.30 Bug Triage Shadow
            # - X # 1.30 Bug Triage Shadow
            # - X # 1.30 Bug Triage Shadow
            # - X # 1.30 Bug Triage Shadow            
```

**Comment by katcosgrove** [comment]
> @katcosgrove @gracenng, how about we merge the existing `ci-signal` and `release-team-bug-triage` GH teams, to a new `release-team-release-signal` team (to align with our new release team structure change, and avoid duplicating same team members across 2 teams)
> 
> ```
>           release-team-release-signal:
>             description: Members of the Release Signal team for the current release cycle.
>             members:
>             - pacoxu # 1.30 Release Signal Lead
>             # - X # 1.30 Bug Triage Shadow
>             # - X # 1.30 Bug Triage Shadow
>             # - X # 1.30 Bug Triage Shadow
>             # - X # 1.30 Bug Triage Shadow            
> ```

Done!

**Pull request merged**